Sharing violation - Chatlog

Every few hours the bot keeps spamming Errors in the cinsole which say (shortened):
Sharing violation on path [path of chatlog] 
at MCC.Chatbot. ChatLog

In my understanding that would mean that another program tries to use the same file on the same time or something about similar. But I dont think there is so it must have to do with a problen coming from MCC. Do you know what is wrong?
Thanks

For some reason it stopped popping up for some days since I wrote the last post. But now it's back again and this is the warning and ther error message which are appearing many times in a row when the error occurs:

17:29:01 [ERROR] OnTextReceived: Got error from MinecraftClient.ChatBots.ChatLog: System.IO.IOException: Sharing violation on path /root/Minecraft/chatlog-MinecraftName-ServerAdress.txt
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x0019e] in <533173d24dae460899d2b10975534bb0>:0
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode) [0x00000] in <533173d24dae460899d2b10975534bb0>:0
  at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode)
  at MinecraftClient.ChatBots.ChatLog.save (System.String tosave) [0x0003d] in <0401b4e711114db0a8c3f277e178e060>:0
  at MinecraftClient.ChatBots.ChatLog.GetText (System.String text) [0x00080] in <0401b4e711114db0a8c3f277e178e060>:0
  at MinecraftClient.McClient+<>c__DisplayClass132_0.<OnTextReceived>b__0 (MinecraftClient.ChatBot bot) [0x00000] in <0401b4e711114db0a8c3f277e178e060>:0
  at MinecraftClient.McClient.DispatchBotEvent (System.Action`1[T] action, System.Collections.Generic.IEnumerable`1[T] botList) [0x00022] in <0401b4e711114db0a8c3f277e178e060>:0

17:29:03 [WARN] OnInternalCommand: Got error from MinecraftClient.ChatBots.ChatLog: System.IO.IOException: Sharing violation on path /root/Minecraft/chatlog-MinecraftName-ServerAdress.txt
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x0019e] in <533173d24dae460899d2b10975534bb0>:0
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode) [0x00000] in <533173d24dae460899d2b10975534bb0>:0
  at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode)
  at MinecraftClient.ChatBots.ChatLog.save (System.String tosave) [0x0003d] in <0401b4e711114db0a8c3f277e178e060>:0
  at MinecraftClient.ChatBots.ChatLog.OnInternalCommand (System.String commandName, System.String commandParams, System.String result) [0x00016] in <0401b4e711114db0a8c3f277e178e060>:0
  at MinecraftClient.McClient.PerformInternalCommand (System.String command, System.String& response_msg, System.Collections.Generic.Dictionary`2[TKey,TValue] localVars) [0x0012a] in <0401b4e711114db0a8c3f277e178e060>:0

--- a/MinecraftClient/ChatBots/ChatLog.cs
+++ b/MinecraftClient/ChatBots/ChatLog.cs
@@ -19,6 +19,7 @@ namespace MinecraftClient.ChatBots
         private bool savePrivate = true;
         private bool saveInternal = true;
         private string logfile;
+        private object logfileLock = new object();
 
         /// <summary>
         /// This bot saves the messages received in the specified file, with some filters and date/time tagging.
@@ -112,16 +113,18 @@ namespace MinecraftClient.ChatBots
         {
             if (dateandtime)
                 tosave = GetTimestamp() + ' ' + tosave;
-
-            string directory = Path.GetDirectoryName(logfile);
-            if (!String.IsNullOrEmpty(directory) && !Directory.Exists(directory))
-                Directory.CreateDirectory(directory);
-            FileStream stream = new FileStream(logfile, FileMode.OpenOrCreate);
-            StreamWriter writer = new StreamWriter(stream);
-            stream.Seek(0, SeekOrigin.End);
-            writer.WriteLine(tosave);
-            writer.Dispose();
-            stream.Close();
+            lock (logfileLock)
+            {
+                string directory = Path.GetDirectoryName(logfile);
+                if (!String.IsNullOrEmpty(directory) && !Directory.Exists(directory))
+                    Directory.CreateDirectory(directory);
+                FileStream stream = new FileStream(logfile, FileMode.OpenOrCreate);
+                StreamWriter writer = new StreamWriter(stream);
+                stream.Seek(0, SeekOrigin.End);
+                writer.WriteLine(tosave);
+                writer.Dispose();
+                stream.Close();
+            }
         }
     }
 }

